go/private/test.bzl

```diff
@@ -66,7 +75,7 @@ def _go_test_impl(ctx):
       library = None,
       want_coverage = False,
       importpath = ctx.label.name + "~testmain~",
-      golibs = [golib],
+      golibs = [golib] + covered_libs,
```

Comment:
It actually does not matter because emit_library_actions immediately collects the libs into depsets internally (direct and transitive)